agent/acs/handler/attach_eni_handler.go

```diff
@@ -85,64 +85,67 @@ func (attachENIHandler *attachENIHandler) handleMessages() {
 	for {
 		select {
 		case message := <-attachENIHandler.messageBuffer:
-			attachENIHandler.handleSingleMessage(message)
+			if err := attachENIHandler.handleSingleMessage(message); err != nil {
+				seelog.Warnf("Unable to handle ENI Attachment message [%s]: %v", message.String(), err)
+			}
 		case <-attachENIHandler.ctx.Done():
 			return
 		}
 	}
 }
 
 // handleSingleMessage acks the message received
-// TODO: Send response to validate ENI attachment
-func (attachENIHandler *attachENIHandler) handleSingleMessage(message *ecsacs.AttachTaskNetworkInterfacesMessage) error {
+func (handler *attachENIHandler) handleSingleMessage(message *ecsacs.AttachTaskNetworkInterfacesMessage) error {
+	receivedAt := time.Now()
 	// Validate fields in the message
-	err := validateAttachTaskNetworkInterfacesMessage(message)
-	if err != nil {
-		return errors.Wrapf(err, "attach eni message handler: error validating AttachTaskNetworkInterfac message received from ECS")
+	if err := validateAttachTaskNetworkInterfacesMessage(message); err != nil {
+		return errors.Wrapf(err,
+			"attach eni message handler: error validating AttachTaskNetworkInterface message received from ECS")
 	}
-
-	err = attachENIHandler.acsClient.MakeRequest(&ecsacs.AckRequest{
+	if err := handler.acsClient.MakeRequest(&ecsacs.AckRequest{
```

Comment:
One concern is that if the make request takes long time, it will block the rest of the process, like start the timer, save the state.